pilot/pkg/model/context.go

```diff
@@ -92,6 +92,13 @@ func (e *Environment) Mesh() *meshconfig.MeshConfig {
 	return nil
 }
 
+func (e *Environment) MeshConfigNamespace() string {
```

Comment:
I don't think the concept of "MeshConfigNamespace" makes sense. We have two existing namespaces:
* System namespace, where istiod runs. Very likely it is wrong to tie any API to us
* Root namespace, defined in mesh config. This is already used by all the other APIs

Do we need to create a new concept, or is root namespace applicable here?

Reply:
It's just the namespace of the mesh config resource, currently the ext_authz service (extension provider) is defined in the MeshConfig and the namespace of the service will be default to the namespace of the containing resource (MeshConfig). 
If later the definition is moved to its own dedicated custom resource, the same rule will still apply to use the namespace of that dedicated resource.

Why is this creating a new concept? Isn't the namespace of MeshConfig already there and we just reuse it anyway.

Root namespace could simplify things here but I'm not sure if it aligns with the purpose of root namespace, I thought it is only for storing global configs but not for deploying actual services and pods?

The other alternative is to ask users to always specify the namespace, but I'm not sure if this is good with @louiscryan and @smawson

Reply:
>  Isn't the namespace of MeshConfig already there and we just reuse it anyway.

Mesh config can also be read from a file - in fact it is in default case, albeit a mounted one. In this code that means it is *always* istio-system, even if the user does not even have an istio-system created at all.

> The other alternative is to ask users to always specify the namespace, but I'm not sure if this is good with 

Strong +1 to this

Reply:
I think users would much rather specify a namespace than deal with the complexity of a default here

Reply:
Additionally, if we do have a default, the mesh config namespace is almost certainly the wrong one for a user. We should not be encouraging users to run their apps in istio-system

Reply:
yeah, that makes sense, especially if the mesh config is loaded from file then it doesn't belong to any specific namespace, updated to require the user to specify the namespace for now.